src/node/internal/internal_dns.ts

```diff
@@ -0,0 +1,264 @@
+// Copyright (c) 2017-2022 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
+// Copyright Joyent and Node contributors. All rights reserved. MIT license.
+
+import { sendDnsRequest } from 'node-internal:internal_dns_client';
+import { DnsError } from 'node-internal:internal_errors';
+import { validateString } from 'node-internal:validators';
+import * as errorCodes from 'node-internal:internal_dns_constants';
+import { default as dnsUtil } from 'node-internal:dns';
+
+export function getServers(): string[] {
+  throw new Error('Not implemented');
+}
+
+export function lookup(): void {
+  throw new Error('Not implemented');
+}
+
+export function lookupService(): void {
+  throw new Error('Not implemented');
+}
+
+export function resolve(): void {
+  throw new Error('Not implemented');
+}
+
+export function resolve4(): void {
+  throw new Error('Not implemented');
+}
+
+export function resolve6(): void {
+  throw new Error('Not implemented');
+}
+
+export function resolveAny(): void {
+  throw new Error('Not implemented');
+}
+
+export function resolveCname(name: string): Promise<string[]> {
+  validateString(name, 'name');
+
+  // Validation errors needs to be sync.
+  // Return a promise rather than using async qualifier.
+  return sendDnsRequest(name, 'CNAME').then((json) => {
+    if (!('Answer' in json)) {
+      // DNS request should contain an "Answer" attribute, but it didn't.
+      throw new DnsError(name, errorCodes.NODATA, 'queryCname');
+    }
+
+    return json.Answer.map((a) => {
+      // Cloudflare DNS returns "nodejs.org." whereas
+      // Node.js returns "nodejs.org" as a CNAME data.
+      return a.data.slice(0, -1);
```

Comment:
To be extra safe here (in case the CF DNS behavior ever changed unexpectedly), it might be worth going ahead and checking to see if the answer endsWith "."